cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -383,10 +383,13 @@ private class Node0 extends Node, TNode0 {
  * An instruction, viewed as a node in a data flow graph.
  */
 class InstructionNode extends Node0 {
-  override InstructionNode0 node;
   Instruction instr;
 
-  InstructionNode() { instr = node.getInstruction() }
+  InstructionNode() {
+    node.(InstructionNode0).getInstruction() = instr
+    or
+    Ssa::getIRRepresentationOfOperand(node.(SingleUseOperandNode0).getOperand()) = instr
```

Comment:
Similarly, here's a case where we had to consider both cases explicitly (since they don't share a common base class).